src/plone/restapi/deserializer/blocks.py

```diff
@@ -38,25 +52,28 @@ def __call__(self, value):
         portal = getMultiAdapter(
             (self.context, self.request), name="plone_portal_state"
         ).portal()
-        portal_url = portal.absolute_url()
-        context_url = self.context.absolute_url()
-        relative_up = len(context_url.split("/")) - len(portal_url.split("/"))
         if self.field.getName() == "blocks":
             for block in value.values():
                 if block.get("@type") == "text":
                     entity_map = block.get("text", {}).get("entityMap", {})
                     for entity in entity_map.values():
                         if entity.get("type") == "LINK":
                             href = entity.get("data", {}).get("url", "")
-                            before = href  # noqa
-                            if href and href.startswith(portal_url):
-                                path = href[len(portal_url) + 1 :].encode("utf8")
-                                uid, suffix = path2uid(portal, path)
-                                if uid:
-                                    href = relative_up * "../" + "resolveuid/" + uid
-                                    if suffix:
-                                        href += suffix
-                                    entity["data"]["href"] = href
-                                    entity["data"]["url"] = href
-                                print("DESERIALIZE " + before + " -> " + href)  # noqa
+                            deserialized_href = path2uid(
+                                context=self.context, portal=portal, href=href
+                            )
+                            entity["data"]["href"] = deserialized_href
+                            entity["data"]["url"] = deserialized_href
+                else:
+                    # standard blocks can have an "url" or "href" field
```

Comment:
@cekk @sneridagh is it fair to assume all blocks have a url/href field? I am just wondering here if we have a unit test that tests what happens if a block does not have those fields. From what I can see in the code blocks this should work...

Reply:
if they don't have these fields, they don't look for them, so it shouldn't be a problem.
I left this part of code untouched (setting both href and url), but probably we could do a better job setting only the needed one and save some bytes

Reply:
I think we are fine.